### ISSUE TITLE
Cannot assign a device for operation 'dense_0/dense/Tensordot/ListDiff'

### ISSUE THREAD

**Issue by dieka13** [tensorflow/tensorflow#17149]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
16.04.3 LTS (Xenial Xerus)
- **TensorFlow installed from (source or binary)**:
pip
- **TensorFlow version (use command below)**:
('v1.5.0-0-g37aa430d84', '1.5.0')
- **Python version**:
2.7.12
- **Bazel version (if compiling from source)**:
not from source
- **GCC/Compiler version (if compiling from source)**:
not from source
- **CUDA/cuDNN version**:
9.0 / 7
- **GPU model and memory**:
GTX 1070 8GB
- **Exact command to reproduce**:
  - loading model through tensorflow.contrib.predictor.from_saved_model
  - loading model through tensorflow serving (built from latest source, with GPU support)

### Describe the problem
I'm trying to load Estimator saved using export_savedmodel.
the model is a speech to text model with CNN, ctc_loss and ctc_greedy_decoder, trained on single GTX 1070. Training run successfully and the model is also saved. but when i load it with tensorflow serving or from_saved_model i got this error:
```
failed: Invalid argument: Cannot assign a device for operation 'dense_0/dense/Tensordot/ListDiff': Could not satisfy explicit device specification '/device:GPU:0' because no supported kernel for GPU devices is available.
Registered kernels:
  device='CPU'; T in [DT_STRING]; out_idx in [DT_INT64]
  device='CPU'; T in [DT_STRING]; out_idx in [DT_INT32]
  device='CPU'; T in [DT_DOUBLE]; out_idx in [DT_INT64]
  device='CPU'; T in [DT_DOUBLE]; out_idx in [DT_INT32]
  device='CPU'; T in [DT_FLOAT]; out_idx in [DT_INT64]
  device='CPU'; T in [DT_FLOAT]; out_idx in [DT_INT32]
  device='CPU'; T in [DT_BFLOAT16]; out_idx in [DT_INT64]
  device='CPU'; T in [DT_BFLOAT16]; out_idx in [DT_INT32]
  device='CPU'; T in [DT_HALF]; out_idx in [DT_INT64]
  device='CPU'; T in [DT_HALF]; out_idx in [DT_INT32]
  device='CPU'; T in [DT_INT8]; out_idx in [DT_INT64]
  device='CPU'; T in [DT_INT8]; out_idx in [DT_INT32]
  device='CPU'; T in [DT_UINT8]; out_idx in [DT_INT64]
  device='CPU'; T in [DT_UINT8]; out_idx in [DT_INT32]
  device='CPU'; T in [DT_INT16]; out_idx in [DT_INT64]
  device='CPU'; T in [DT_INT16]; out_idx in [DT_INT32]
  device='CPU'; T in [DT_UINT16]; out_idx in [DT_INT64]
  device='CPU'; T in [DT_UINT16]; out_idx in [DT_INT32]
  device='CPU'; T in [DT_INT32]; out_idx in [DT_INT64]
  device='CPU'; T in [DT_INT32]; out_idx in [DT_INT32]
  device='CPU'; T in [DT_INT64]; out_idx in [DT_INT64]
  device='CPU'; T in [DT_INT64]; out_idx in [DT_INT32]
```

### Source code / logs
[error.txt](https://github.com/tensorflow/tensorflow/files/1739607/error.txt)
[source code](https://github.com/tensorflow/tensorflow/files/1739608/bug.zip) (run estimator_CNN.py)

**Comment by bignamehyp** [comment]
what if you use tf.einsum? 
Similar to issue #11619

**Comment by dieka13** [comment]
@bignamehyp: 
I'm not calling tf.tensordot explicitly in my code

**Comment by isaprykin** [comment]
I think the session created by SavedModelPredictor needs to have allow_soft_placement=True.  I'll reach out to the author internally, they don't seem to be on github.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @isaprykin: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by david-haber** [comment]
@dieka13 tf.layers.dense produces a tensordot op if its input is of rank higher than 2. see [here](https://github.com/tensorflow/tensorflow/blob/r1.8/tensorflow/python/layers/core.py#L156).

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 30 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by dmonn** [comment]
Same problem with `tf.contrib.slim.fully_connected`

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 16 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 31 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by jamieas** [comment]
@dieka13 Can you try

`tf.contrib.predictor.from_saved_model(
  saved_model_dir, config=tf.ConfigProto(allow_soft_placement=True))`

**Comment by dieka13** [comment]
@jamieas sorry, it's been too long. I do not longer have access to the source code.

**Comment by jamieas** [comment]
@dieka13 thanks for your response. Sorry that it took so long for this bug to be addressed.

@rohan100jain should we close this bug as obsolete?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.
